### ISSUE TITLE
Use a deferred object to get rdio and musixmatch data

### ISSUE THREAD

**Issue by ryanjadhav** [RepublicOfKids/SingAlongWithMe#19]
shoot those api requests in parallel rather than serially. 

refer to http://api.jquery.com/deferred.promise/

**Comment by ryanjadhav** [comment]
Currently musixmatch fires with a callback to query rdios API. This is terrible and slower than it should be.

**Comment by ryanjadhav** [comment]
http://api.jquery.com/jQuery.when/
